CMakeLists.txt

```diff
@@ -271,10 +279,14 @@ add_compile_options(-DconfigUSE_NEWLIB_REENTRANT=1)
 add_compile_options(-g)
 
 # optimizations
-if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  add_compile_options(-Og)
+if(MCU STREQUAL "STM32F429VI")
+  add_compile_options(-O0)
 else()
-  add_compile_options(-Os)
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    add_compile_options(-Og)
+  else()
+    add_compile_options(-Os)
+  endif()
```

Comment:
/question/ Is this intentional? I would expect that a configuration with MCU=STM32F429VI and CMAKE_BUILD_TYPE=Release would produce a binary with `-Os` (as that is usually expected from a release configuration) and `-O0` would be used for that MCU only when CMAKE_BUILD_TYPE is Debug. In other words, something like

```
if (build type is Debug)
  -O0 if mcu is F429VI else -Og
else
  -Os
```